src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -29,6 +29,7 @@ std::shared_ptr<ov::Node> change_constant_precision_to_fp16(std::shared_ptr<ov::
     for (size_t i = 0; i < size; ++i) {
         // if it is smaller than the smallest positive normal fp16 but not zero
         if (std::abs(src_data[i]) <= ov::float16::from_bits(0x0400) && src_data[i] != 0.0f) {
+            dst_data[i] = static_cast<ov::float16>(src_data[i]);
```

Comment:
Will it be casted to zero anyway here? Shouldn't we have smallest fp16 value here?

Reply:
I was also surprised that it indeed will not be casted to zero. I checked that.

BTW, i forgot to add unittest, will do it :)

Reply:
It might mean that `0x0400` is not really smallest fp16 number. As far as I can see here https://en.wikipedia.org/wiki/Half-precision_floating-point_format smallest is actually `0x0001`

Reply:
I knew that `0x0004` was not the smallest fp16, it was the smallest normal value (looks like a reflected that in the comment line above). For some reason i forgot to change that :(

Now I indeed use the smallest fp16, which is `0x0001`, as you stated. Added unit-test for this fix as well. Please take a look